src/whitenoise/accounts/relays.rs

```diff
@@ -154,152 +154,124 @@ impl Whitenoise {
-            .fetch_relays_from(list, account.pubkey, RelayType::Nostr)
-            .await?;
-        for relay in existing_relay_list {
-            account.nip65_relays.insert(relay);
+    pub async fn add_relay_to_account(
+        &self,
+        pubkey: PublicKey,
+        relay: RelayUrl,
+        relay_type: RelayType,
+    ) -> Result<()> {
+        if !self.logged_in(&pubkey).await {
+            return Err(WhitenoiseError::AccountNotFound);
         }
 
-        self.update_account_relays_db(
-            &account.pubkey,
-            account.nip65_relays.clone().into_iter().collect(),
-            RelayType::Nostr,
-        )
-        .await?;
+        let account = self.get_account(&pubkey).await?;
 
-        self.publish_relay_list_for_account(account, RelayType::Nostr)
-            .await
-    }
+        // Add the relay to the appropriate relay set
+        match relay_type {
+            RelayType::Nostr => {
+                account.nip65_relays.insert(relay.clone());
+            }
+            RelayType::Inbox => {
+                account.inbox_relays.insert(relay.clone());
+            }
+            RelayType::KeyPackage => {
+                account.key_package_relays.insert(relay.clone());
+            }
+        }
 
-    /// Add relay to the inbox relays list and establishes the connection
-    /// Update the relay list in the database
-    pub async fn add_inbox_relay(&self, account: &Account, relay: RelayUrl) -> Result<()> {
-        self.nostr.connect_to_relay(relay.clone()).await?;
-        account.inbox_relays.insert(relay);
-        self.update_account_relays_db(
-            &account.pubkey,
-            account.inbox_relays.clone().into_iter().collect(),
-            RelayType::Inbox,
-        )
-        .await?;
-        self.publish_relay_list_for_account(account, RelayType::Inbox)
-            .await
-    }
+        // Save the updated account
+        self.save_account(&account).await?;
 
-    /// Add relay to the inbox relays list, does not establish the connection
-    /// Update the relay list in the database
-    pub async fn add_key_package_relay(&self, account: &Account, relay: RelayUrl) -> Result<()> {
-        account.key_package_relays.insert(relay);
-        self.update_account_relays_db(
-            &account.pubkey,
-            account.key_package_relays.clone().into_iter().collect(),
-            RelayType::KeyPackage,
-        )
-        .await?;
-        self.publish_relay_list_for_account(account, RelayType::KeyPackage)
-            .await
+        // Update the in-memory account
+        {
+            let mut accounts = self.write_accounts().await;
+            accounts.insert(account.pubkey, account.clone());
+        }
+
+        // Ensure relays are connected
+        self.nostr
+            .ensure_relays_connected(DashSet::from_iter([relay]))
+            .await?;
+
+        self.publish_relay_list_for_account(&account, relay_type, &None)
+            .await?;
```

Comment:
Should we make sure that we're connected to all relays or would that overkill?

If so, should we make that always when calling `publish_relay_list_for_account`? So basically maybe have that function handle it within?